app/src/main/kotlin/com/waz/zclient/feature/backup/usecase/CreateBackUpUseCase.kt

```
@@ -5,17 +5,21 @@ import com.waz.zclient.core.exception.FeatureFailure
 import com.waz.zclient.core.functional.Either
 import com.waz.zclient.core.usecase.UseCase
 import com.waz.zclient.feature.backup.BackUpRepository
+import kotlinx.coroutines.*
 
-class CreateBackUpUseCase(private val backUpRepositories: List<BackUpRepository>) : UseCase<Unit, Unit> {
+class CreateBackUpUseCase(
+    private val backUpRepositories: List<BackUpRepository>,
+    private val coroutineScope: CoroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Default)
+) : UseCase<Unit, Unit> {
 
     override suspend fun run(params: Unit): Either<Failure, Unit> {
         var failure = false
-        //TODO we can speed it up by async + await maybe?
-        for (repo in backUpRepositories) {
-            if (repo.backUp().isLeft) {
-                failure = true
-                break
-            }
+        coroutineScope.launch {
```

I'll be looking at this again to further optimise this for performance improvements including adding a fail fast approach which it currently lacks.